### ISSUE TITLE
Support relative (in addition to absolute) `min_delta` parameters in `keras.callbacks.ReduceLROnPlateau`

### ISSUE THREAD

**Issue by ruancomelli** [keras-team/keras#15863]
## System information

TensorFlow version (you are using): 2.7.0
Are you willing to contribute it (Yes/No): *YES*

## Describe the feature and the current behavior/state

I am reopening the issue https://github.com/tensorflow/tensorflow/issues/33675 to assess community interest in that feature and to discuss possible implementations.
Quoting the original issue, the requested feature can be described as:
> Regarding tf.keras.callbacks.ReduceLROnPlateau: The min_delta parameter is currently an absolute number which indicates when a meaningful reduction in the monitored loss has accrued. It makes no sense to use an absolute number for two reasons -
> 1. Every loss has a different dynamic range and hence a different definition for a meaningful reduction
> 2. A "meaningful reduction" decreases as the training progresses. The higher the epoch the smaller of a change in loss is expected.
For these two reasons I think that a percentage of change in the monitored loss is much more useful.

I do not have access to the code that I used anymore, but the problem that I was trying to solve was:
> [...] the loss is ~1e5 at the beginning of training, while the goal is to achieve a loss as low as 10 at the end. One can easily see that min_delta=10 has very different meanings in the beginning and in the end.

I was able to solve this by implementing a custom version of `ReduceLROnPlateau` that accepted relative `min_delta`s.

Just for the record, [PyTorch supports this feature](https://pytorch.org/docs/stable/generated/torch.optim.lr_scheduler.ReduceLROnPlateau.html#reducelronplateau) by accepting the parameters `threshold` (equivalent to Kera's `min_delta`) and `threshold_mode`, that specifies whether `threshold` should be considered an absolute or a relative change.

## Will this change the current API? How?

Yes: a new parameter should be added to the initializer of `ReduceLROnPlateau`. This addition can be done in a backward-compatible manner with a sensible choice of default values.

## Who will benefit from this feature?

Anyone who uses the `ReduceLROnPlateau` callback, especially people working with models whose loss varies a lot during training.

## [Contributing](https://github.com/keras-team/keras/blob/master/CONTRIBUTING.md)

- Do you want to contribute a PR? (yes/no): *YES*
- If yes, please read [this page](https://github.com/keras-team/keras/blob/master/CONTRIBUTING.md) for instructions
- Briefly describe your candidate solution(if contributing):

I currently have two candidate solutions:
1. **add a new parameter `min_delta_mode: Literal['absolute', 'relative']`**: passing `min_delta_mode='absolute'` (the default behavior) instructs Keras to consider `min_delta` as an absolute change, as in the current behavior; passing `min_delta_mode='relative'` instructs Keras to consider `min_delta` as a relative change.
2. **add a new parameter `min_delta_rel: Optional[float]`**: the user must pass either `min_delta` or `min_delta_rel` (but not both) - passing `min_delta` is the current option; passing `min_delta_rel` achieves the new behavior.

Note that both candidates are equivalent, it's just a matter of choosing the best interface. Supposing that we choose option 1, the [`ReduceLROnPlateau._reset`](https://github.com/keras-team/keras/blob/4b9a1410d126fa818a7245aeb90bcebed6398048/keras/callbacks.py#L2713) method would be changed so that `self.monitor_op` is defined depending on `self.mode` and `self.min_delta_mode` according to the following table:

| `mode` | `min_delta_mode` | `monitor_op` |
| - | - | - |
| `'min'` | `'absolute'` | `lambda current, best: np.less(current, best - self.min_delta)` |
| `'max'` | `'absolute'` | `lambda current, best: np.greater(current, best + self.min_delta)` |
| `'min'` | `'relative'` | `lambda current, best: np.less(current, (1 - self.min_delta)*best)` |
| `'max'` | `'relative'` | `lambda current, best: np.greater(current, (1 + self.min_delta)*best)` |

**Comment by chenmoneygithub** [comment]
@rchao Rick, could you take a closer look? Thanks!
